docarray/doc_index/backends/elastic_doc_index.py

```diff
@@ -0,0 +1,484 @@
+        if int(self._server_version.split('.')[0]) >= 8:
+            os.environ['ELASTIC_CLIENT_APIVERSIONING'] = '1'
+
+        # ElasticSearh index setup
+        self._index_vector_params = ('dims',)
+
+        body: Dict[str, Any] = {
+            'mappings': {
+                'dynamic': True,
+                '_source': {'enabled': 'true'},
+                'properties': {},
+            }
+        }
+
+        for col_name, col in self._column_infos.items():
+            # if not col.config:
+            #     continue  # do not create column index if no config is given
+            body['mappings']['properties'][col_name] = self._create_index(col)
+
+        if self._client.indices.exists(index=self._index_name):
+            self._client.indices.put_mapping(
+                index=self._index_name, body=body['mappings']
+            )
+        else:
+            self._client.indices.create(index=self._index_name, body=body)
+
+        if len(self._db_config.index_settings):
+            self._client.indices.put_settings(
+                index=self._index_name, body=self._db_config.index_settings
+            )
+
+        self._refresh(self._index_name)
+
+    ###############################################
+    # Inner classes for query builder and configs #
+    ###############################################
+
+    class QueryBuilder(BaseDocumentIndex.QueryBuilder):
+        def __init__(self, outer_instance, **kwargs):
+            super().__init__()
+            self._outer_instance = outer_instance
+            self._query: Dict[str, Any] = {
+                'query': defaultdict(lambda: defaultdict(list))
+            }
+
+        def build(self, *args, **kwargs) -> Any:
+            if (
+                'script_score' in self._query['query']
+                and 'bool' in self._query['query']
+                and len(self._query['query']['bool']) > 0
+            ):
+                self._query['query']['script_score']['query'] = {}
+                self._query['query']['script_score']['query']['bool'] = self._query[
+                    'query'
+                ]['bool']
+                del self._query['query']['bool']
+
+            return self._query
+
+        def find(
+            self,
+            query: Union[AnyTensor, BaseDocument],
+            search_field: str = 'embedding',
+            limit: int = 10,
+        ):
+            if isinstance(query, BaseDocument):
+                query_vec = BaseDocumentIndex._get_values_by_column(
+                    [query], search_field
+                )[0]
+            else:
+                query_vec = query
+            query_vec_np = BaseDocumentIndex._to_numpy(self._outer_instance, query_vec)
+            self._query['size'] = limit
+            self._query['query']['script_score'] = {
+                'query': {'match_all': {}},
+                'script': {
+                    'source': f'cosineSimilarity(params.query_vector, \'{search_field}\') + 1.0',
```

Comment:
Just a thought: the awesome thing about script score query is that we can make multiple vector similarity comparisons in one query. Do you think we can enable this here too? (Or maybe this is possible and I am missing it.) For example, I have a query with multiple embeddings and I want to compare each to a specific field in my index. Perhaps we can also make this a follow-up ticket.

Reply:
Yeah it's possible. But now the `find` function only accepts one vector. Maybe we can discuss about this feature in the future.